tests/test_cargo_publish.rs

```diff
@@ -84,7 +176,8 @@ test!(simple {
 });
 
 test!(git_deps {
-    let p = project("foo")
+    let root = paths::root().join("git_deps");
+    let p = repo(&root)
```

Comment:
I had some back and forth on this too. There will be errors if someone has 
`
home - git 
-- overall project - hg
---- sub project that is a crate - hg
`

I think a git in git might be more common. 
`
- open source project - top git level
-- osx libs 
---- ruby
---- rust
----- crate code
`